src/prototypo-canvas.js

```diff
@@ -502,18 +528,151 @@ PrototypoCanvas.prototype.setupEvents = function( pCanvasInstance ) {
+							);
+						}
+						pCanvasInstance.typographicFrame.snappingHelper.strokeColor = '#FF4AFF';
+						pCanvasInstance.typographicFrame.snappingHelper.opacity = 0.5;
+						pCanvasInstance.typographicFrame.snappingHelper.applyMatrix = false;
+						pCanvasInstance.shiftLock.isLineDrawn = true;
+						if (!this.isShiftPressed) {
+							this.changesToConfirm = cursors;
+							sendManualChanges(cursors);
+						}
+						break;
+					case 'snapped':
+						if (pCanvasInstance.snapping.axis === 'y') {
+							pCanvasInstance.typographicFrame.snappingHelper.segments[0].point.x =  !pCanvasInstance.shiftLock.isLocked ?
+							this.selectedSegment.point.x + event.delta.x :
+							this.selectedSegment.point.x;
+						}
+						if (pCanvasInstance.snapping.axis === 'x') {
+							pCanvasInstance.typographicFrame.snappingHelper.segments[0].point.y = !pCanvasInstance.shiftLock.isLocked ?
+							this.selectedSegment.point.y - event.delta.y :
+							this.selectedSegment.point.y;
+						}
+						cursors = pCanvasInstance.snapping.axis === 'y' ?
+						{
+							[`contours.${contourIdx}.nodes.${nodeIdx}.x`]: event.delta.x,
+							[`contours.${contourIdx}.nodes.${nodeIdx}.y`]: 0,
+						} :
+						{
+							[`contours.${contourIdx}.nodes.${nodeIdx}.x`]: 0,
+							[`contours.${contourIdx}.nodes.${nodeIdx}.y`]: -event.delta.y,
+						};
+						if (!this.isShiftPressed) {
+							this.changesToConfirm = cursors;
+							sendManualChanges(cursors);
+						}
+						break;
+					case 'unSnapped':
+						cursors = pCanvasInstance.snapping.axis === 'y' ?
+						{
+							[`contours.${contourIdx}.nodes.${nodeIdx}.x`]: event.delta.x,
+							[`contours.${contourIdx}.nodes.${nodeIdx}.y`]: yDirection === 'bottom' ?
+																														-unSnappingTrigger :
+																														unSnappingTrigger,
+						} :
+						{
+								[`contours.${contourIdx}.nodes.${nodeIdx}.x`]: xDirection === 'right' ?
+																															unSnappingTrigger :
+																															-unSnappingTrigger,
+								[`contours.${contourIdx}.nodes.${nodeIdx}.y`]: -event.delta.y,
+						};
+						pCanvasInstance.snapping.deltaX = 0;
+						pCanvasInstance.snapping.deltaY = 0;
+						pCanvasInstance.snapping.isSnapped = false;
+						pCanvasInstance.snapping.axis = '';
+						pCanvasInstance.snapping.snappedTo = undefined;
+						pCanvasInstance.typographicFrame.snappingHelper.remove();
+						if (!this.isShiftPressed) {
+							this.changesToConfirm = cursors;
+							sendManualChanges(cursors);
+						}
+						break;
+					default:
+						cursors = {
+							[`contours.${contourIdx}.nodes.${nodeIdx}.x`]: event.delta.x,
+							[`contours.${contourIdx}.nodes.${nodeIdx}.y`]: -event.delta.y,
+						};
+						if (!this.isShiftPressed) {
+							this.changesToConfirm = cursors;
+							sendManualChanges(cursors);
+						}
+						break;
+				}
+				break;
+			default:
+				break;
+		}
+		if (this.isShiftPressed) {
```

Comment:
Why not consider ShiftPressed another type of change and put it inside the switch ?